server/routes/calculationQuestionRoutes.test.ts

```diff
@@ -502,6 +502,22 @@ describe('Calculation question routes tests', () => {
       })
   })
 
+  it('POST /calculation/:nomsId/reason should return to the reason page and display the error message if the other reason is selected and more than 40 characters been entered', () => {
+    prisonerService.getPrisonerDetail.mockResolvedValue(stubbedPrisonerData)
+    calculateReleaseDatesService.getCalculationReasons.mockResolvedValue(stubbedCalculationReasons)
+    config.featureToggles.calculationReasonToggle = true
+
+    return request(app)
+      .post('/calculation/A1234AA/reason/')
+      .type('form')
+      .send({ calculationReasonId: ['11'], otherReasonDescription: 'A string which is at least 40 characters' })
+      .expect(200)
+      .expect(res => {
+        expect(res.text).toContain('You must enter less than 40 characters')
+        expectMiniProfile(res.text, expectedMiniProfile)
```

Comment:
If I understand correctly this is also trying to keep the other text filled in case of error. Maybe add an expectation here for that? Otherwise LGTM